### DISCUSSION TITLE
Return JSON array

### DISCUSSION THREAD

**Discussion by KayJay89** [fastapi/fastapi#7495] [Questions] [answered]
Hello everybody,

First of all this is a phenomenal framework and I am loving what I have learnt so far. I'm fairly new to this one so I do apologize in advance if the below has been asked already.

I'm trying to return a JSON array on one of my routes but I am failing to do so as I am getting below error:

```
pydantic.error_wrappers.ValidationError: 1 validation error for Management
response
  value is not a valid dict (type=type_error.dict)
```
My endpoint:

```
@app.get("/management/{ticker}", response_model=Management)
async def get_management_data(
    ticker: str = Path(
        ...,
        title="Ticker of the company's data you want to get",
        min_length=1,
        max_length=5,
        regex="^[a-zA-Z]+$",
    )
):
    management_data = Ticker(ticker).all_modules
    response_data = []
    for officer in management_data[ticker]["assetProfile"]["companyOfficers"]:
        manager_data = {}
        for key, value in officer.items():
            if key in ["name", "title", "age", "totalPay"]:
                manager_data.update({key: value})
        response_data.append(manager_data)
    return response_data
```

So the basic idea is that the api user inputs a 1 to 5 letter long string, the API then looks up the company's details and returns a list of Manager models, both of which are defined as follows:

```
class Manager(BaseModel):
    name: str
    title: str
    age: int
    salary: int = None


class Management(BaseModel):
    manager: List[Manager]
```
So basically what I would expect to return is:

```
{
    {"name": "Name of manager", "title": "CEO", "age": 39,},
    {"name": "Name of manager", "title": "CFO", "age": 56,},
    {"name": "Name of manager", "title": "CMO", "age": 44,},
}
```

I suppose I am doing something wrong but I have not found the answer after looking for hours online and trying out different things. I would very much appreciate any hint or help to get me unstuck. 

KR,
Kevin

**Comment by KayJay89** [comment, marked as answer]
And of course, right when I post it I find the solution. For those in similar situation, here is what I did:

1.Delete completely:
```
class Management(BaseModel):
    manager: List[Manager]
```


2.Update the response_model as follows:

**Old:**
`response_model=Management`

**New:**
`response_model=List[Manager]`

**Reference to docs where this was mentioned:**
https://fastapi.tiangolo.com/tutorial/extra-models/#list-of-models

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
